host/sgx/create.c

```diff
@@ -202,8 +200,8 @@ static oe_result_t _add_control_pages(
     oe_result_t result = OE_UNEXPECTED;
     oe_page_t* page = NULL;
 
-    if (!context || !enclave_addr || !enclave_size || !entry || !vaddr ||
-        !enclave)
+    if (!context || !enclave->start_address || !enclave->size || !entry ||
+        !vaddr || !enclave)
```

Comment:
```suggestion
    if (!context || !enclave || !enclave->start_address || !enclave->size ||
         !entry || !vaddr)
```
BUG: move the !enclave check before dereferencing it